### PR TITLE
fix the first issue

### PR DESCRIPTION
**Pull request by huangliang992**

### CODE REVIEW

**Review by yaronyam — COMMENTED** [review]
Great to see this implemented as an Intention. Great contribution.
Posted few comments.

Found some issues while testing....

1. I get the below error in IDEA when opening Intentions and expanding right side options. if it's not feasible to support preview mode - override this method for now
```java
@Override  
public @NotNull IntentionPreviewInfo generatePreview(@NotNull Project project, @NotNull Editor editor, @NotNull PsiFile file) {  
    return IntentionPreviewInfo.EMPTY;  
}
```
2. when cursor on method. invoking TestMe popup, not via the Intention, in order to create an additional test class - updates an existing test class with additional method. also got occational errors in such scenario. not sure they were related.

---------

```
  Exceptions occurred on invoking the intention 'TestMe: Generate test for method' on a copy of the file.
  
  com.intellij.codeInsight.intention.impl.preview.IntentionPreviewUnsupportedOperationException: It's unexpected to invoke this method on an intention preview calculating.
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewEditor.notImplemented(IntentionPreviewEditor.kt:15)
at com.intellij.openapi.editor.impl.ImaginaryEditor.getContentComponent(ImaginaryEditor.java:84)
at com.intellij.ui.popup.AbstractPopup.showInBestPositionFor(AbstractPopup.java:707)
at com.weirddev.testme.intellij.ui.popup.TestMePopUpHandler.show(TestMePopUpHandler.java:142)
at com.weirddev.testme.intellij.ui.popup.TestMePopUpHandler.invoke(TestMePopUpHandler.java:53)
at com.weirddev.testme.intellij.intention.CreateTestForMethodIntention.invoke(CreateTestForMethodIntention.java:64)
at com.intellij.codeInsight.intention.PsiElementBaseIntentionAction.invoke(PsiElementBaseIntentionAction.java:41)
at com.intellij.codeInsight.intention.IntentionAction.generatePreview(IntentionAction.java:111)
at com.intellij.codeInsight.intention.impl.config.IntentionActionWrapper.generatePreview(IntentionActionWrapper.java:90)
at com.intellij.codeInsight.intention.impl.IntentionActionWithTextCaching$MyIntentionAction.generatePreview(IntentionActionWithTextCaching.java:242)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$3$lambda$2$lambda$1(IntentionPreviewComputable.kt:124)
at com.intellij.model.SideEffectGuard.computeWithAllowedSideEffects(SideEffectGuard.java:25)
at com.intellij.model.SideEffectGuard.computeWithoutSideEffects(SideEffectGuard.java:18)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$3$lambda$2(IntentionPreviewComputable.kt:123)
at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.lambda$postponeFormattingInside$2(PostprocessReformattingAspectImpl.java:140)
at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.postponeFormattingInside(PostprocessReformattingAspectImpl.java:149)
at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.postponeFormattingInside(PostprocessReformattingAspectImpl.java:139)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$3(IntentionPreviewComputable.kt:122)
at com.intellij.codeInsight.intention.preview.IntentionPreviewUtils.previewSession(IntentionPreviewUtils.java:127)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview(IntentionPreviewComputable.kt:121)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.generatePreview(IntentionPreviewComputable.kt:89)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.tryCreateDiffContent(IntentionPreviewComputable.kt:65)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.call(IntentionPreviewComputable.kt:43)
at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.call(IntentionPreviewComputable.kt:38)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$OTelMonitor.callWrapped(NonBlockingReadActionImpl.java:832)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$OTelMonitor$MonitoredComputation.call(NonBlockingReadActionImpl.java:864)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.insideReadAction(NonBlockingReadActionImpl.java:602)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$attemptComputation$4(NonBlockingReadActionImpl.java:565)
at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1133)
at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:76)
at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:63)
at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:133)
at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$1(ProgressIndicatorUtils.java:114)
at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:111)
at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:76)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.attemptComputation(NonBlockingReadActionImpl.java:565)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$1(NonBlockingReadActionImpl.java:466)
at com.intellij.openapi.application.impl.NonBlockingReadActionImpl$Submission.lambda$transferToBgThread$2(NonBlockingReadActionImpl.java:481)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
at java.base/java.lang.Thread.run(Thread.java:833)
```

File: src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+     */
+    public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element)
+        throws IncorrectOperationException {
+        PsiMethod method = (PsiMethod)element.getParent();
+        final PsiClass srcClass = CreateTestMeAction.getContainingClass(element);
+        // if is not testable show a error dialog
+        if (!MethodFactory.isTestable(method, srcClass)) {
+            String message = "Method \"" + method.getName() + "\" is not testable!";
+            String title = "TestMe: Method Not Testable";
+            Messages.showErrorDialog(message, title);
```
Comment (yaronyam) [inline src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java:60]: I like the idea of popping an error notification in such case 👍 
but currently it results in an error:
```
java.lang.Throwable: AWT events are not allowed inside write action: java.awt.event.FocusEvent[FOCUS_LOST,temporary,opposite=EditorComponent
at com.intellij.openapi.ui.Messages.showErrorDialog(Messages.java:636)  
at com.weirddev.testme.intellij.intention.CreateTestForMethodIntention.invoke(CreateTestForMethodIntention.java:60)
```
instead use error hint, UX also seems a bit more friendly. see JB implementation for a very similar use case - https://github.com/JetBrains/intellij-community/blob/e0370874cac76407163f9f5ad68f479ac03c173f/java/java-impl/src/com/intellij/testIntegration/createTest/GenerateMissedTestsAction.java#L67

furthermore, I think it would be good to re use some fast checks as in `GenerateMissedTestsAction.isAvailable`  from file referenced above . like verify method is public and not abstract

File: src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        } else {
+            new TestMeActionHandler(true).invoke(project, editor, element.getContainingFile());
```
Comment (yaronyam) [inline src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java:62]: would probably be better to pass the method element already resolved by IDEA rather then pass the file and try to resolve it by cursor location.  we can create another method for this use case on TestMeActionHandler

Comment (yaronyam) [reply src/main/java/com/weirddev/testme/intellij/intention/CreateTestForMethodIntention.java:62]: Furthermore, there's still a bug when trying to generate another test class (not via intention for method) for a class that already has a test, if cursor is on a test method - existing test is updated instead of popping a window to select alternative test class name as in previous behavior. there are also other exceptions and errors. my point is, that now that method generation is explicitly triggered by user via intention - we need to take advantage of this and separate logic in both CreateTestMeAction and TestMeActionHandler for these 2 user scenarios . this includes changes already been merged in previous PR.

**Review by yaronyam — APPROVED** [review]
Great progress! much more stable now. found 1 small issue, there's a logged exception (PSB) after generating test for method when target UT exists but is empty. could not reproduce this after the first exception. if you can't reproduce nevermind.

Summarizing main issues left:
- User cannot pick the relevant UT in case more then 1 exists. 
- Generating UT for method - will add a UT in test class with default name if such exists, regardless of language compatibility (perhaps not a significant issue if assuming user should be responsible picking the relevant language). this happens even if user has an existing additional UT in required language but not in default name.  it's an edge cases, though I'm familiar with some projects that transitioned to UT in Groovy and have legacy tests in java...

To resolve both these issues, would be great to extend this feature in the future and open a popup chooser for selecting target UT class in case more then 1 exists. for now we can settle with current functionality, it's still a great feature!


*Logged exception after generating test for method when target UT exists but is empty (not reproducible consistently):
```
2024-05-24 16:53:09,917 [2489813] SEVERE - #c.i.p.i.s.t.CompositeElement - Assertion failed: anchorBefore == null || anchorBefore.getTreeParent() == parent  
java.lang.Throwable: Assertion failed: anchorBefore == null || anchorBefore.getTreeParent() == parent  
    at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:465)  
    at com.intellij.psi.impl.source.tree.CompositeElement.addChild(CompositeElement.java:561)  
    at com.intellij.psi.impl.source.tree.CompositeElement.addChildren(CompositeElement.java:663)  
    at com.intellij.psi.impl.source.codeStyle.CodeEditUtil.addChildren(CodeEditUtil.java:53)  
    at com.intellij.psi.impl.source.tree.CompositeElement.addInternal(CompositeElement.java:438)  
    at com.intellij.psi.impl.source.tree.java.ImportListElement.addInternal(ImportListElement.java:47)  
    at com.intellij.psi.impl.source.JavaStubPsiElement.addAfter(JavaStubPsiElement.java:72)  
    at com.weirddev.testme.intellij.utils.TestFileUpdateUtil.updateTestFile(TestFileUpdateUtil.java:78)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator.createTestClassFromCodeTemplate(TestMeGenerator.java:133)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator.createTestClass(TestMeGenerator.java:110)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator$1$1.compute(TestMeGenerator.java:66)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator$1$1.compute(TestMeGenerator.java:61)  
    at com.intellij.openapi.application.impl.ApplicationImpl.lambda$runWriteAction$9(ApplicationImpl.java:1001)  
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteActionWithClass(ApplicationImpl.java:980)  
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1001)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator$1.compute(TestMeGenerator.java:61)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator$1.compute(TestMeGenerator.java:59)  
    at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.postponeFormattingInside(PostprocessReformattingAspectImpl.java:149)  
    at com.weirddev.testme.intellij.generator.TestMeGenerator.generateTest(TestMeGenerator.java:59)  
    at com.weirddev.testme.intellij.action.CreateTestMeAction$1.run(CreateTestMeAction.java:134)  
    at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:219)  
    at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:174)  
    at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:164)  
    at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:150)  
    at com.weirddev.testme.intellij.action.CreateTestMeAction.createTest(CreateTestMeAction.java:131)  
    at com.weirddev.testme.intellij.action.TestMeCreator.invoke(TestMeCreator.java:36)  
    at com.weirddev.testme.intellij.action.TestMeCreator.createTest(TestMeCreator.java:26)  
    at com.weirddev.testme.intellij.action.TestMeAdditionalAction.execute(TestMeAdditionalAction.java:53)  
    at com.weirddev.testme.intellij.ui.popup.TestMePopUpHandler.lambda$show$1(TestMePopUpHandler.java:127)  
    at com.intellij.openapi.ui.popup.PopupChooserBuilder.lambda$setItemChosenCallback$0(PopupChooserBuilder.java:207)
```

**Comment by yaronyam** [comment]
Released in EAP release - [version 6.5.0.eap](https://plugins.jetbrains.com/plugin/9471-testme/versions/eap/545342) . Will be included in upcoming stable release

**Comment by yaronyam** [comment]
released in [v6.5.0](https://plugins.jetbrains.com/plugin/download?rel=true&updateId=546453)
